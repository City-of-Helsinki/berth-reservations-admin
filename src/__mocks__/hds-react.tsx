import React from 'react';
import { DateInputProps } from 'hds-react';

// Replace DateInput implementation because it increases test completion time
// by tends of seconds.
//
// This DateInput doesn't have the calendar component which is the root cause
// for the performance issues.
export const FastDateInput = ({ onChange, value, label, id, name }: DateInputProps) => {
  return (
    <label htmlFor={id}>
      {label}
      <input
        type="text"
        id={id}
        name={name}
        value={value ?? ''}
        onChange={(e: React.ChangeEvent<HTMLInputElement>) => {
          const value = e.target.value;

          onChange?.(value, new Date());
        }}
      />
    </label>
  );
};

// Re-export everything
export * from 'hds-react';

// Override DateInput
export { FastDateInput as DateInput };
