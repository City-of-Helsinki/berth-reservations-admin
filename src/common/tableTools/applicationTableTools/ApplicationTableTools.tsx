import React from 'react';
import { useTranslation } from 'react-i18next';
import { TextInput } from 'hds-react';

import { ApplicationStatus } from '../../../@types/__generated__/globalTypes';
import Select from '../../select/Select';
import { APPLICATION_STATUS } from '../../utils/constants';
import styles from './applicationTableTools.module.scss';

interface ApplicationTableToolsProps {
  count?: number;
  nameFilter?: string;
  statusFilter?: ApplicationStatus;
  onNameFilterChange(nameFilter: string | undefined): void;
  onStatusFilterChange(statusFilter?: ApplicationStatus): void;
}

const ApplicationTableTools = ({
  count,
  nameFilter,
  onNameFilterChange,
  statusFilter,
  onStatusFilterChange,
}: ApplicationTableToolsProps) => {
  const { t } = useTranslation();
  const options = [
    { label: t('common.all'), value: '' as ApplicationStatus },
    ...Object.values(ApplicationStatus).map((status) => {
      return { label: t(APPLICATION_STATUS[status].label), value: status };
    }),
  ];
  return (
    <div className={styles.container}>
      <span>{t('applicationStateTableTools.count', { count: count ?? 0 })}</span>
      <div className={styles.filtersContainer}>
        <TextInput
          id="applicationStateTableTools_nameFilter"
          label={t('applicationStateTableTools.nameFilterLabel')}
          onChange={(e) => onNameFilterChange((e.target as HTMLInputElement).value || undefined)}
          value={nameFilter ?? ''}
        />
        <Select
          label={t('applicationStateTableTools.selectLabel')}
          className={styles.select}
          onChange={(e) =>
            (e.target.value as string) === '' ? onStatusFilterChange(undefined) : onStatusFilterChange(e.target.value)
          }
          value={statusFilter ?? ('' as ApplicationStatus)}
          options={options}
          visibleOptions={options.length + 1}
        />
      </div>
    </div>
  );
};

export default ApplicationTableTools;
