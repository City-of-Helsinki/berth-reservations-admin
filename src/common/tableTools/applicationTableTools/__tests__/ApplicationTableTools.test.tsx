import React from 'react';
import { shallow } from 'enzyme';
import { TextInput } from 'hds-react';
import ApplicationTableTools from '../ApplicationTableTools';

describe('ApplicationTableTools', () => {
  it('renders correctly', () => {
    const wrapper = shallow(
      <ApplicationTableTools count={1} onNameFilterChange={jest.fn()} onStatusFilterChange={jest.fn()} />
    );
    expect(wrapper.render()).toMatchSnapshot();
  });

  it('calls onNameFilterChange when text is typed into the name text input', () => {
    const nameToFilter = 'Testi Testinen';

    const onNameFilterChangeMock = jest.fn();
    const wrapper = shallow(
      <ApplicationTableTools count={1} onNameFilterChange={onNameFilterChangeMock} onStatusFilterChange={jest.fn()} />
    );

    wrapper.find(TextInput).simulate('change', { target: { value: nameToFilter } });
    expect(onNameFilterChangeMock).toBeCalledWith(nameToFilter);

    // Assert calls with undefined when empty string is input
    wrapper.find(TextInput).simulate('change', { target: { value: '' } });
    expect(onNameFilterChangeMock).toBeCalledWith(undefined);
  });
});
