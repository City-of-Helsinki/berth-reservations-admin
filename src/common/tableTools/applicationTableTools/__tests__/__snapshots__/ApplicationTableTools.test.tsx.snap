// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ApplicationTableTools renders correctly 1`] = `
<div
  class="container"
>
  <div
    class="leftContainer"
  >
    <span>
      Tuloksien lukumäärä: 1
    </span>
  </div>
  <div
    class="filtersContainer"
  >
    <div
      class="TextInput-module_root__2CMNr text-input_hds-text-input__2LODq"
    >
      <label
        class="FieldLabel-module_label__1zrXK "
        for="applicationStateTableTools_nameFilter"
      >
        Asiakkaan nimi
      </label>
      <div
        class="TextInput-module_inputWrapper__3Rvel text-input_hds-text-input__input-wrapper__1OqYG"
      >
        <input
          class="TextInput-module_input__1BlHi text-input_hds-text-input__input__GJm5C"
          id="applicationStateTableTools_nameFilter"
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      class="Select-module_root__Ka5uO select"
    >
      <label
        class="FieldLabel-module_label__1zrXK "
        for="hds-select-1-toggle-button"
        id="hds-select-1-label"
      >
        Hakemuksen tila
      </label>
      <div
        class="Select-module_wrapper__1WQXs"
      >
        <button
          aria-expanded="false"
          aria-haspopup="listbox"
          aria-labelledby="hds-select-1-label hds-select-1-toggle-button"
          aria-owns="hds-select-1-menu"
          class="Select-module_button__1aIsm"
          id="hds-select-1-toggle-button"
          type="button"
        >
          <span
            class="Select-module_buttonLabel__1fqu5"
          >
            Kaikki
          </span>
          <svg
            aria-hidden="true"
            class="Icon-module_icon__1Jtzj icon_hds-icon__1YqNC Icon-module_s__2WGWe icon_hds-icon--size-s__2Lkik Select-module_angleIcon__2-9AD"
            role="img"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g
              fill="none"
              fill-rule="evenodd"
            >
              <path
                d="M0 0h24v24H0z"
              />
              <path
                d="M12 13.5l5-5 1.5 1.5-6.5 6.5L5.5 10 7 8.5z"
                fill="currentColor"
              />
            </g>
          </svg>
        </button>
        <ul
          aria-labelledby="hds-select-1-label"
          class="Select-module_menu__1H2aU"
          id="hds-select-1-menu"
          role="listbox"
          style="max-height:468px"
          tabindex="-1"
        />
      </div>
    </div>
  </div>
</div>
`;
